python/shotgun_fields/date_widget.py

```diff
@@ -30,5 +55,70 @@ def _string_value(self, value):
         :param value: The value to convert into a string
         :type value: A String representing the date in YYYY-MM-DD form
         """
-        dt = datetime.datetime.strptime(value, "%Y-%m-%d")
-        return self._create_human_readable_timestamp(dt)
+        if not isinstance(value, datetime.date):
+            value = datetime.datetime.strptime(value, "%Y-%m-%d").date()
+        return self._create_human_readable_date(value)
+
+
+class DateEditorWidget(QtGui.QDateEdit):
+    """
+    Allows editing of a ``date`` field value as returned by the Shotgun API.
+
+    Pressing ``Enter`` or ``Return`` when the widget has focus will cause the
+    value to be applied and the ``value_changed`` signal to be emitted.
+    """
+    __metaclass__ = ShotgunFieldMeta
+    _EDITOR_TYPE = "date"
+
+    def get_value(self):
+        """
+        :return: The internal value being displayed by the widget.
+        """
+        value = self.date()
+
+        if hasattr(QtCore, "QVariant"):
+            # pyqt
+            return value.toPyDate()
+        else:
+            # pyside
+            return value.toPython()
+
+    def keyPressEvent(self, event):
+        """
+        Provides shortcuts for applying modified values.
+
+        :param event: The key press event object
+        :type event: :class:`~PySide.QtGui.QKeyEvent`
+        """
+        if event.key() in [QtCore.Qt.Key_Enter, QtCore.Qt.Key_Return]:
```

Comment:
These particular "apply" key shortcuts show up in 6 or so of the line edit based widgets. Other, text edit based widgets, use different keys. Other widgets don't use key events to apply values. To me it feels like there's not enough similarity across all the widgets to put something in the metaclass. I'm happy to though if you have something you think is reasonable.